### PR TITLE
feat: Add support for creating a security group for VPC endpoint(s)

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Add support for creating a security group for VPC endpoint(s)

## Motivation and Context
- VPC endpoints require a security group, and most of the interface endpoint services use https/443 - this change lets users makes these requirements easier for users when utilizing the VPC endpoint sub-module while also allowing them to extend its use

## Breaking Changes
- No

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -103,23 +112,6 @@ module "vpc_endpoints" {
       policy          = data.aws_iam_policy_document.dynamodb_endpoint_policy.json
       tags            = { Name = "dynamodb-vpc-endpoint" }
     },
-    ssm = {
```
Comment (bryantbiggs) [inline examples/complete/main.tf:106]: we don't need all of these - they just make the example take longer and cost more

File: modules/vpc-endpoints/variables.tf [hunk 2]
```diff
@@ -39,3 +39,43 @@ variable "timeouts" {
...
 }
+
+################################################################################
+# Security Group
+################################################################################
+
+variable "create_security_group" {
+  description = "Determines if a security group is created"
+  type        = bool
+  default     = false
```
Comment (bryantbiggs) [inline modules/vpc-endpoints/variables.tf:50]: default is `false` so we don't surprise users (like I did with the ALB module 🙃)

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 5.1.0 :tada:

**Comment by segoja7** [comment]
Hi @bryantbiggs 

I think that the new feature to create security groups for vpc endpoints is not working.

```
module "vpc_endpoints" {
  source   = "terraform-aws-modules/vpc/aws//modules/vpc-endpoints"
  version  = "5.1.0"
  vpc_id   = data.aws_vpc.vpc_datasource.id
  endpoints = {
    logs = {
      create              = false
      service             = "logs"
      private_dns_enabled = true
      create_security_group      = true
      security_group_description = "security group for cloudwatchlogs-vpc-endpoint"
      security_group_name_prefix = "cloudwatchlogs-vpc-endpoint-sg"
      security_group_rules  = {
      ingress = {
        from_port   = 443
        to_port     = 443
        protocol    = "tcp"
        description = "from 1.2.3.4/5:443"
        cidr_blocks = "0.0.0.0/0"
      }
      egress = {
        from_port   = -1
        to_port     = -1
        protocol    = "-1"
        description = "Allow all outgoing traffic"
        cidr_blocks = "0.0.0.0/0"
      }
    }
      subnet_ids = [for subnet_ids in data.aws_subnet.example : subnet_ids.id]
      tags       = { Name = "cloudwatchlogs-vpc-endpoint" }
    }
  }
}
```

This is my ouput in the console:

`  # module.vpc_endpoints.aws_vpc_endpoint.this["logs"] will be created
  + resource "aws_vpc_endpoint" "this" {
      + arn                   = (known after apply)
      + cidr_blocks           = (known after apply)
      + dns_entry             = (known after apply)
      + id                    = (known after apply)
      + ip_address_type       = (known after apply)
      + network_interface_ids = (known after apply)
      + owner_id              = (known after apply)
      + policy                = (known after apply)
      + prefix_list_id        = (known after apply)
      + private_dns_enabled   = true
      + requester_managed     = (known after apply)
      + route_table_ids       = (known after apply)
      + security_group_ids    = (known after apply)
      + service_name          = "com.amazonaws.us-east-1.logs"
      + state                 = (known after apply)
      + subnet_ids            = [
          + "subnet-0120e7db456cdea23",
          + "subnet-01ee56dc688436975",
          + "subnet-082e2b84b329c0e7b",
          + "subnet-0b7088527fc8e9850",
          + "subnet-0c970355eb251ebc1",
          + "subnet-0df47a8ae0643cfbe",
        ]
      + tags                  = {
          + "Name" = "cloudwatchlogs-vpc-endpoint"
        }
      + tags_all              = {
          + "Name" = "cloudwatchlogs-vpc-endpoint"
        }
      + vpc_endpoint_type     = "Interface"
      + vpc_id                = "vpc-07bd49a034b4dfc44"

      + timeouts {
          + create = "10m"
          + delete = "10m"
          + update = "10m"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.`

**Comment by bryantbiggs** [comment]
your syntax is incorrect

**Comment by segoja7** [comment]
Sorry for that, can you explain to me where is the error?

**Comment by bryantbiggs** [comment]
How have you tried checking/validating your code?

**Comment by segoja7** [comment]
My apologies, now is workink.

`module "vpc_endpoints" {
  source   = "terraform-aws-modules/vpc/aws//modules/vpc-endpoints"
  version  = "5.1.0"
  vpc_id   = data.aws_vpc.vpc_datasource.id
  create_security_group = true
  create              = true
  #private_dns_enabled = true
  security_group_description = "security group for logs-vpc-endpoint"
  security_group_name_prefix = "logs-vpc-endpoint-sg"
  security_group_rules  = {
      ingress = {
        from_port   = 443
        to_port     = 443
        protocol    = "tcp"
        description = "from 1.2.3.4/5:443"
        cidr_blocks = ["0.0.0.0/0"]
      }
      egress = {
        from_port   = -1
        to_port     = -1
        protocol    = "-1"
        description = "Allow all outgoing traffic"
        cidr_blocks = ["0.0.0.0/0"]
      }
    }
  endpoints = {
    logs = {
      service    = "logs"
      subnet_ids = [for subnet_ids in data.aws_subnet.example : subnet_ids.id]
      tags       = { Name = "logs-vpc-endpoint" }
    }
  }
}`

**Pull request merged**